File: DesingPatterns/Pizza/Pizzaria.java
package Pizza;

class Pizzaria {

    public void fazerPizzas() {
        PizzaFactory[] factories = {new PizzaBrasileira(), new PizzaItaliana()};
        
        for (PizzaFactory factory : factories) {
            Pizza pizza = factory.criarPizza();
            PizzaVeg pizzaVeg = factory.criarPizzaVeg();
            pizzaVeg.preparar();
            pizza.servir(pizzaVeg);
        }
    }

    public static void main(String[] args) {
        Pizzaria pizzaria = new Pizzaria();
        pizzaria.fazerPizzas();
        System.out.println("Estudantes: \n");
        System.out.println ("Jair Eugênio Ferreira - 202401791 \n");
        System.out.println ("João Vitor Sousa de Oliveira - 202401794 \n");
        System.out.println ("Lucas Almeida Oliveira Isaac - 202401802 \n");
        System.out.println ("Lucca Magnino - 202401805 \n");
   
    }
}

File: DesingPatterns/pizza_py/Pizzaria.py

from PizzaBrasileira import PizzaBrasileira
from PizzaItaliana import PizzaItaliana
from Pizza import Pizza
from PizzaVeg import PizzaVeg

# Cliente
class Pizzaria:

    def fazer_pizzas(self):
        for factory in [PizzaBrasileira(), PizzaItaliana()]:
            self.factory = factory
            self.pizza = self.factory.criar_pizza()
            self.pizza_veg = self.factory.criar_pizza_veg()
            self.pizza_veg.preparar()
            self.pizza.servir(self.pizza_veg)


pizzaria = Pizzaria()
pizzaria.fazer_pizzas()
print ("Estudantes: \n")
print ("Jair Eugênio Ferreira - 202401791 \n")
print ("João Vitor Sousa de Oliveira - 202401794 \n")
print ("Lucas Almeida Oliveira Isaac - 202401802 \n")
print ("Lucca Magnino - 202401805 \n")